config/routes.rb

```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'upload/' => 'upload#index', as: 'upload'
```

Comment:
Why do you need to create multiple views? Each method in that will have it's own view